prediction_market_agent/db/pinecone_handler.py

```diff
@@ -0,0 +1,55 @@
+import base64
+import typing as t
+from typing import Optional
+
+from langchain_openai import OpenAIEmbeddings
+from langchain_pinecone import PineconeVectorStore
+from pinecone import Pinecone
+
+from prediction_market_agent.agents.think_thoroughly_agent.models import (
+    PineconeMetadata,
+)
+from prediction_market_agent.utils import APIKeys
+
+INDEX_NAME = "omen-markets"
+
+
+# ToDo - Move to PMAT
+class PineconeHandler:
```

Comment:
If `PineconeHandler` is our class, why do we need type-ignores above in the code?

Reply:
mypy has trouble realizing that self.init_pinecone() initializes `self.pinecone_handler`, thus should not be of type `PineconeHandler|None` anymore, but rather `PineconeHandler`.
Hence the type ignore.